packages/compiler/core/cli.ts

```diff
@@ -274,10 +279,17 @@ async function getCompilerOptions(args: {
   import?: string[];
   watch?: boolean;
   "diagnostic-level": string;
+  "service-code-path"?: string;
 }): Promise<CompilerOptions> {
   // Ensure output path
   const outputPath = resolve(args["output-path"]);
+  const serviceCodePath = args["service-code-path"]
```

Comment:
similarly, I think all of this code preparing the directory needs to be in the generator itself